Firestore/Source/Core/FSTSyncEngine.mm

```diff
@@ -307,26 +308,35 @@ - (void)transactionWithRetries:(int)retries
     workerQueue->Enqueue(
         [self, retries, workerQueue, updateCallback, resultCallback, transaction, maybe_result] {
           if (!maybe_result.ok()) {
+            if (retries > 0 && [self isRetryableTransactionError:maybe_result.status()] &&
+                !transaction->IsPermanentlyFailed()) {
+              workerQueue->VerifyIsCurrentQueue();
```

Comment:
This seems overly paranoid since we just enqueued this code onto the worker queue 5 lines above.  Since there are no callbacks, etc. going on, it would be pretty crazy for this to somehow fail.